lib/openstudio-standards/schedules/information.rb

```diff
@@ -4,12 +4,346 @@ module OpenstudioStandards
+      end
+
+      result = { 'min' => vals.min, 'max' => vals.max }
+
+      return result
+    end
+
+    # Returns an array of average hourly values from a ScheduleCompact object
+    # Returns 8760 values, 8784 for leap years.
+    #
+    # @param schedule_compact [OpenStudio::Model::ScheduleCompact] OpenStudio ScheduleCompact object
+    # @return [Array<Double>] Array of hourly values for the year
+    def self.schedule_compact_get_hourly_values(schedule_compact)
+      # set a ScheduleTypeLimits if none is present
+      # this is required for the ScheduleTranslator instantiation
+      unless schedule_compact.scheduleTypeLimits.is_initialized
+        schedule_type_limits = OpenStudio::Model::ScheduleTypeLimits.new(model)
+        schedule_compact.setScheduleTypeLimits(schedule_type_limits)
+      end
+
+      # convert to a ScheduleRuleset and use its method
+      sch_translator = ScheduleTranslator.new(schedule_compact.model, schedule_compact)
+      schedule_ruleset = sch_translator.convert_schedule_compact_to_schedule_ruleset
+      result = OpenstudioStandards::Schedules.schedule_ruleset_get_hourly_values(schedule_ruleset)
+
+      return result
+    end
+
+    # @!endgroup Information:ScheduleCompact
+
+    # @!group Information:ScheduleDay
+
+    # Returns the ScheduleDay daily equivalent full load hours (EFLH).
+    #
+    # @param schedule_day [OpenStudio::Model::ScheduleDay] OpenStudio ScheduleDay object
+    # return [Double] The daily total equivalent full load hours for this schedule
+    def self.schedule_day_get_equivalent_full_load_hours(schedule_day)
+      daily_flh = 0
+      values = schedule_day.values
+      times = schedule_day.times
+
+      previous_time_decimal = 0
+      times.each_with_index do |time, i|
+        time_decimal = (time.days * 24.0) + time.hours + (time.minutes / 60.0) + (time.seconds / 3600.0)
+        duration_of_value = time_decimal - previous_time_decimal
+        daily_flh += values[i] * duration_of_value
+        previous_time_decimal = time_decimal
+      end
+
+      return daily_flh
+    end
+
+
+    # Returns an array of average hourly values from a ScheduleDay object
+    # Returns 24 values
+    #
+    # @param schedule_day [OpenStudio::Model::ScheduleDay] OpenStudio ScheduleDay object
+    # @return [Array<Double>] Array of hourly values for the day
+    def self.schedule_day_get_hourly_values(schedule_day)
+      schedule_values = []
+
+      # determine smallest time interval
+      times = schedule_day.times
+      time_interval_min = 15.0
+      previous_time_decimal = 0.0
+      times.each_with_index do |time, i|
+        time_decimal = (time.days * 24.0 * 60.0) + (time.hours * 60.0) + time.minutes + (time.seconds / 60)
+        interval_min = time_decimal - previous_time_decimal
+        time_interval_min = interval_min if interval_min < time_interval_min
+        previous_time_decimal = time_decimal
+      end
+      time_interval_min = time_interval_min.round(0).to_i
+
```

Comment:
Just a few more checks here might worth considering:
`0<time_interval_min && 60%time_interval_min == 0 for even division`.

Reply:
can't force an even division, because the average won't be correct in some cases with non-factors of 60. Consider a time interval of 7 minutes - can't use a lower number without returning the wrong average.
```
(7..60).step(7).to_a
=> [7, 14, 21, 28, 35, 42, 49, 56]
```

The time_interval_min being > zero is enforced by times being sequential in the schedule_day object.